### ISSUE TITLE
Properly constructing a (PaymentSigningKeyShelley_ed25519) signing key from Cardano CLI CBOR

### ISSUE THREAD

**Issue by Montel98** [Emurgo/cardano-serialization-lib#340]
I have a Plutus script that I can successfully submit transactions with using the serialization-lib, except when logic is added to check for a private key signature:
```
txSignedByApp :: Bool
txSignedByApp = txSignedBy info $ appPubKeyHash
```
I would like to parse the signing key file generated by the CLI with the serialization lib. In the CLI, I can simply use the corresponding `payment.skey` to sign the transaction. I have attempted to construct a `PrivateKey` type by reading the `payment.skey` cbor hex and parsing it as follows:

`const signingKey = Cardano.PrivateKey.from_normal_bytes(cbor.decodeFirstSync(signingKeyHexCBOR));` 

where `signingKeyHexCBOR` is the utf-8 encoded hex string. No errors are emitted by the serialization-lib but the transaction fails validation. 

Any pointers on how to correctly construct a signing key would be greatly appreciated. Thanks!

**Comment by rooooooooob** [comment]
@Montel98 My first thought was since it's called `HexCBOR` that that could be it since we don't use CBOR encodings for keys, hashes or addresses (they are `raw bytes` instead of CBOR i.e. `CBORBYTES(len) || raw bytes` or `CBORBYTES(indefinite) || raw bytes || CBORBREAK`) but I think that would fail parsing if those got confused (due to a length mismatch) so maybe cardano-cli does the same thing for us with the bytes but just calls them CBOR to match with the rest.

Let's check to see if it's CBOR or not. CBOR bytes would be `0x58` as first byte for 24+ length bytestrings in definite or `0x5f` for indefinite. If it's not one of those then it's likely not related to the above at all so you can ignore that.

>No errors are emitted by the serialization-lib but the transaction fails validation

Can you elaborate? Is the pubkey included as a plutus datum? We have #313 which will be resolved by #317 which can result in conflicting datum hashes (#313 reported that one of the cases where this happens involved a key stored in a datum).

**Comment by Montel98** [comment]
@rooooooooob Thanks for the reply, yes the first byte is 0x58. I can confirm that the private key is in fact being derived correctly. Passing the hex string through a cbor decoder OR removing the first 2 bytes does not throw an error, but directly passing the hex (as expected) gives an `invalid normal key` parsing error from the serialization lib.

It turns out the issue is actually caused by an empty `signatories []` in the error log. I didn't realise that `required signatures` needs to be explicitly called to embed the pub key hash into the transaction.
